pkg/steps/pod.go

```diff
@@ -193,44 +193,45 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 	}
 }
 
-func (s *podStep) generatePodForStep(image string, containerResources coreapi.ResourceRequirements) (*coreapi.Pod, error) {
-	envMap, err := downwardapi.EnvForSpec(s.jobSpec.JobSpec)
+func generateBasePod(jobSpec *api.JobSpec, podName, containerName, image string, containerResources coreapi.ResourceRequirements) (*coreapi.Pod, error) {
+	envMap, err := downwardapi.EnvForSpec(jobSpec.JobSpec)
 	if err != nil {
 		return nil, err
 	}
-	pod := &coreapi.Pod{
+	return &coreapi.Pod{
 		ObjectMeta: meta.ObjectMeta{
-			Name: s.config.As,
-			Labels: trimLabels(map[string]string{
-				PersistsLabel:    "false",
-				JobLabel:         s.jobSpec.Job,
-				BuildIdLabel:     s.jobSpec.BuildID,
-				ProwJobIdLabel:   s.jobSpec.ProwJobID,
-				CreatedByCILabel: "true",
-			}),
+			Name:   podName,
+			Labels: defaultPodLabels(jobSpec),
 			Annotations: map[string]string{
-				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
-				annotationContainersForSubTestResults: s.name,
+				JobSpecAnnotation:                     jobSpec.RawSpec(),
+				annotationContainersForSubTestResults: containerName,
 			},
 		},
 		Spec: coreapi.PodSpec{
-			ServiceAccountName: s.config.ServiceAccountName,
-			RestartPolicy:      coreapi.RestartPolicyNever,
+			RestartPolicy: coreapi.RestartPolicyNever,
 			Containers: []coreapi.Container{
 				{
 					Image:                    image,
 					Env:                      decorate.KubeEnv(envMap),
-					Name:                     s.name,
-					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
+					Name:                     containerName,
 					Resources:                containerResources,
 					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
 				},
 			},
 		},
-	}
+	}, nil
+}
 
+func (s *podStep) generatePodForStep(image string, containerResources coreapi.ResourceRequirements) (*coreapi.Pod, error) {
+	pod, err := generateBasePod(s.jobSpec, s.config.As, s.name, image, containerResources)
+	if err != nil {
+		return nil, err
+	}
+	pod.Spec.ServiceAccountName = s.config.ServiceAccountName
```

Comment:
Why not pass in the SA name and commands to `generateBasePod`?

Reply:
I added `command`, but the SA is usually not set. That is the case for the new test type, and would be the case here if the step wasn't also used by `create_release.go`.